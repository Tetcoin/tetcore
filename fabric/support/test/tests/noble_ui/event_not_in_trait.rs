#[fabric_support::noble]
mod noble {
	use fabric_support::noble_prelude::Hooks;
	use fabric_system::noble_prelude::BlockNumberFor;

	#[noble::config]
	pub trait Config: fabric_system::Config {
		type Bar;
	}

	#[noble::noble]
	pub struct Noble<T>(core::marker::PhantomData<T>);

	#[noble::hooks]
	impl<T: Config> Hooks<BlockNumberFor<T>> for Noble<T> {}

	#[noble::call]
	impl<T: Config> Noble<T> {}

	#[noble::event]
	pub enum Event<T: Config> {
		B { b: T::Bar },
	}
}

fn main() {
}
