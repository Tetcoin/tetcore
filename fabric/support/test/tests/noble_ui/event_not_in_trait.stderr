error: Invalid usage of Event, `Config` contains no associated type `Event`, but enum `Event` is declared (in use of `#[noble::event]`). An Event associated type must be declare on trait `Config`.
 --> $DIR/event_not_in_trait.rs:1:1
  |
1 | #[fabric_support::noble]
  | ^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
